src/main/java/seedu/address/logic/commands/gradetrackercommands/EditAssignmentCommand.java

```diff
@@ -1,16 +1,157 @@
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "CS2101 "
+            + PREFIX_ASSIGNMENT_NAME + "Oral Presentation 2";
+
+    public static final String MESSAGE_EDIT_ASSIGNMENT_SUCCESS = "Edited assignment: \n%1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_MODULE_INVALID = "The module to edit assignment is invalid.";
+    public static final String MESSAGE_DUPLICATE_ASSIGNMENT = "This assignment already exists in the gradetracker.";
+
+    private final Logger logger = LogsCenter.getLogger(EditAssignmentCommand.class);
+
+    private final Index index;
+    private final ModuleName moduleName;
+    private final EditAssignmentDescriptor editAssignmentDescriptor;
+
+    /**
+     * @param index of the assignment in the gradetracker of the module
+     * @param editAssignmentDescriptor details to edit the assignment with
+     */
+    public EditAssignmentCommand(Index index, ModuleName moduleName,
+                                 EditAssignmentDescriptor editAssignmentDescriptor) {
+        requireNonNull(index);
+        requireNonNull(moduleName);
+        requireNonNull(editAssignmentDescriptor);
+
+        logger.info("Editing assignment at position " + index.toString() + " from:" + moduleName.toString() + "");
+        this.index = index;
+        this.moduleName = moduleName;
+        this.editAssignmentDescriptor = new EditAssignmentDescriptor(editAssignmentDescriptor);
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return null;
+        requireNonNull(model);
+        List<Module> lastShownList = model.getFilteredModuleList();
+        Module moduleWithAssignment = null;
+
+        for (Module eachModule : lastShownList) {
+            if (eachModule.getName().equals(moduleName)) {
+                moduleWithAssignment = eachModule;
+                break;
+            }
+        }
+
+        if (moduleWithAssignment == null) {
+            throw new CommandException(MESSAGE_MODULE_INVALID);
+        }
+
+        if (index.getZeroBased() >= moduleWithAssignment.getGradeTracker().getAssignments().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_ASSIGNMENT_DISPLAYED_INDEX);
+        }
+
+        Assignment assignmentToEdit = moduleWithAssignment.getGradeTracker().getSortedAssignments()
+                .get(index.getZeroBased());
+        Assignment editedAssignment = createEditedAssignment(assignmentToEdit, editAssignmentDescriptor);
+
+        if (!assignmentToEdit.isSameAssignment(editedAssignment)
+                && moduleWithAssignment.getGradeTracker().containsDuplicateAssignment(editedAssignment)) {
+            throw new CommandException(MESSAGE_DUPLICATE_ASSIGNMENT);
+        }
+
+        moduleWithAssignment.getGradeTracker().removeAssignment(assignmentToEdit);
```

Comment:
If I am not wrong, the way ModulistList, ContactList and TodoList edit their respective components would be to use a setModule, setContact, setTask method? Maybe to ensure consistency we could use a setAssigment method as well.

Reply:
Thanks for the comments, I'll update accordingly.